File: server/src/main/java/org/dan/jadalnia/app/ws/FestivalListenersProvider.java
package org.dan.jadalnia.app.ws;

import org.dan.jadalnia.app.festival.pojo.Fid;
import org.springframework.context.annotation.Bean;

import javax.inject.Provider;
import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;

public class FestivalListenersProvider implements Provider<Map<Fid, FestivalListeners>> {
    @Bean
    public Map<Fid, FestivalListeners> get() {
        return new ConcurrentHashMap<>();
    }
}

File: server/src/main/java/org/dan/jadalnia/app/ws/FestivalListenersProvider.kt
package org.dan.jadalnia.app.ws

//import java.util.Map
import org.dan.jadalnia.app.festival.pojo.Fid
import org.springframework.context.annotation.Bean

//import javax.inject.Provider
import java.util.concurrent.ConcurrentHashMap

class FestivalListenersProvider {
    @Bean
    public fun getMap(): MutableMap<Fid, FestivalListeners> {
        return ConcurrentHashMap() //<Fid, FestivalListeners>() as Map<Fid, FestivalListeners>)
    }
}
